apps/settings/test/unit/modules/bluetooth/bluetooth_adapter_manager_test.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+suite('BluetoothAdapterManager', function() {
+  var adapterManager;
+
+  setup(function(done) {
+    var modules = [
+      'modules/bluetooth/bluetooth_adapter_manager'
+    ];
+
+    var map = {
+      'modules/bluetooth': {
+        'modules/navigator/mozBluetooth': 'MockNavigatorBluetooth'
+      }
+    };
+
+    this.MockNavigatorBluetooth = {
+      defaultAdapter: {},
+      callbacks: {
+        'onattributechanged': []
+      },
+      addEventListener: function(eventName, callback) {
+        this.callbacks[eventName] = callback;
```

Comment:
Because `callbacks.onattributechanged` is an array, you should push the callback to it. After the change, in the test case we should trigger the callback using `this.callbacks[eventName][0]`.